src/api/contact-center/buckets/buckets.js

```diff
@@ -6,80 +6,48 @@ import eventBus from "../../../utils/eventBus";
 const fieldsToSend = ['domainId', 'name', 'description'];
 
-export const getBucketsList = async (page = 0, size = 10, search) => {
-    const domainId = store.state.userinfo.domainId || undefined;
-    const defaultObject = {
-        _isSelected: false,
-    };
-    if(search && search.slice(-1) !== '*') search += '*';
+const defaultItemObject = {
+    name: '',
+    _dirty: false,
+};
 
-    try {
-        const response = await bucketService.searchBucket(page, size, search, domainId);
-        if (response.items) {
-            return response.items.map(item => {
-                return {...defaultObject, ...item};
-            });
-        }
-        return [];
-    } catch (err) {
-        throw err;
-    }
+const listGetter = new WebitelSDKListGetter(bucketService.searchBucket);
+const itemGetter = new WebitelSDKItemGetter(bucketService.readBucket, defaultItemObject);
+const itemCreator = new WebitelSDKItemCreator(bucketService.createBucket, fieldsToSend);
+const itemUpdater = new WebitelSDKItemUpdater(bucketService.updateBucket, fieldsToSend);
+const itemDeleter = new WebitelSDKItemDeleter(bucketService.deleteBucket);
+
+export const getBucketsList = async (page = 0, size = 10, search) => {
+    return await listGetter.getList({page, size, search});
 };
 
 export const getBucket = async (id) => {
-    const domainId = store.state.userinfo.domainId || undefined;
-    const defaultObject = {
-        name: '',
-        _dirty: false,
-    };
-
-    try {
-        const response = await bucketService.readBucket(id, domainId);
-        return {...defaultObject, ...response};
-    } catch (err) {
-        throw err;
-    }
+    return await itemGetter.getItem(id);
 };
 
 export const addBucket = async (item) => {
-    let itemCopy = deepCopy(item);
-    itemCopy.domainId = store.state.userinfo.domainId || undefined;
-    sanitizer(itemCopy, fieldsToSend);
-    try {
-        const response = await bucketService.createBucket(itemCopy);
-        eventBus.$emit('notificationInfo', 'Sucessfully added');
-        return response.id;
-    } catch (err) {
-        throw err;
-    }
+    return await itemCreator.createItem(item);
 };
 
 export const updateBucket = async (id, item) => {
-    let itemCopy = deepCopy(item);
-    itemCopy.domainId = store.state.userinfo.domainId || undefined;
-    sanitizer(itemCopy, fieldsToSend);
-    try {
-        await bucketService.updateBucket(id, itemCopy);
-        eventBus.$emit('notificationInfo', 'Sucessfully updated');
-    } catch (err) {
-        throw err;
-    }
+    return await itemUpdater.updateItem(id, item);
 };
 
 export const deleteBucket = async (id) => {
-    const domainId = store.state.userinfo.domainId || undefined;
-    try {
-        await bucketService.deleteBucket(id, domainId);
-    } catch (err) {
-        throw err;
-    }
+    return await itemDeleter.deleteItem(id);
 };
 
 export const getBucketPermissions = async (id, page = 0, size = 10, search) => {
```

Comment:
why not to create listGetter instance for this api?

Reply:
+ patcher

Reply:
done